lib/kramdown_rpf/rpf.rb

```diff
@@ -172,6 +202,79 @@ def self.convert_task_to_html(task)
         HEREDOC
       end
 
+      module KnowledgeQuiz
+        def self.convert_feedback_to_html(feedback, index = nil)
+          id = 'feedback'
+          id += "-for-choice-#{index + 1}" unless index.nil?
+          <<~HEREDOC
+            <li class="knowledge_quiz__feedback-item" id="#{id}">
+            #{::Kramdown::Document.new(feedback.strip, KRAMDOWN_OPTIONS).to_html.strip}
+            </li>
+          HEREDOC
+        end
+  
+        def self.convert_label_to_html(label, index, checked)
+          number = index + 1
+          <<~HEREDOC
+            <label for="choice-#{number}">#{::Kramdown::Document.new(label, KRAMDOWN_OPTIONS).to_html.strip}</label>
+            <input type="radio" name="answer" value="#{number}" id="choice-#{number}" #{checked ? 'checked': ''}/>
+          HEREDOC
+        end
+  
+        def self.convert_choices_to_html(text)
+          choices = text.split(CHOICE_BLOCK_REGEXP)
+          choice_html = ''
+          feedback_html = ''
+  
+          single_feedback_match = SINGLE_FEEDBACK_REGEXP.match(choices[0])
+          unless single_feedback_match.nil?
+            feedback_html += convert_feedback_to_html(single_feedback_match[1].strip, nil)
+            choices.shift
+          end
+
+          choices.each.with_index do |choice, index|
+            choice_match = RADIO_REGEXP.match(choice)
+
+            next unless choice_match && choice_match['text']
```

Comment:
You can use safe navigation here (although it is a bit weird with the array operator `[]`)

```suggestion
            next unless choice_match&.[]('text')
```